pkg/token/token.go

```diff
@@ -564,14 +567,16 @@ func (v tokenVerifier) Verify(token string) (*Identity, error) {
 	req.Header.Set(clusterIDHeader, v.clusterID)
 	req.Header.Set("accept", "application/json")
 
+	stsRegion := getStsRegion(parsedURL.Host)
```

Comment:
This is not safe to parse before executing the request. We need to validate this is only parsed after a successful signature response.

Reply:
ah I see we've already done host verification. Move this call to just after that to make it more clear.

Also:
* Can you make your function more robust against bad inputs? (ex: hosts without a period ) I know there is host verification before this, but bad things can happen and your code here should not depend on  ordering if we're going to log/metric this stuff before request execution
* Can you write a test for your `getStsRegion` function?